### ISSUE TITLE
Make ButtonBar.alignment configurable from AlertDialog

### ISSUE THREAD

**Issue by kenzieschmoll** [flutter/flutter#69708]
`ButtonBar` has a configurable `alignment` property. This is a request to expose a property `actionsAlignment` from the `AlertDialog` constructor so that `ButtonBar.alignment` can be set here: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/dialog.dart#L518

**Comment by yash1200** [comment]
I would like to work on this issue.

**Comment by HansMuller** [comment]
We'd prefer to eliminate ButtonBar altogether. Some progress has been made already:

Use OverflowBar instead of ButtonBar in DateRangePicker #62695
Use OverflowBar instead of ButtonBar in DatePicker #62686
Use OverflowBar instead of ButtonBar in TimePicker #62601
Use OverflowBar instead of ButtonBar in MaterialBanner #62703

**Comment by HansMuller** [comment]
More progress around removing ButtonBar:

Removed ButtonBar from AlertDialog #75676

**Comment by HansMuller** [comment]
Last Flutter SDK reference removed: 

Removed ButtonBar from the Scaffold and About widgets #76246

**Comment by HansMuller** [comment]
@kenzieschmoll - The AlertDialog's internal ButtonBar is now gone and the OverflowBar widget that replaces it is not based on a Row. So there's no easy way to introduce a MainAxisAlignment valued property into AlertDialog.  On the other hand, it would be easy to add an AlignmentDirectional valued property to AlertDialog which would be used to horizontally align the OverflowBar within dialog. Currently this alignment value is hardwired to  AlignmentDirectional.centerEnd.  Would that serve your use case?

**Comment by pedromassangocode** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by ycherniavskyi** [comment]
@HansMuller my use case is broken exactly because of hardcoded `AlignmentDirectional.centerEnd`.
Previously I use the next `Theme` property to center actions within `AlertDialog`:
```dart
buttonBarTheme: ButtonBarThemeData(
  alignment: MainAxisAlignment.center,
),
```
But after eliminate `ButtonBar` usage and hardcoded actions container alignment I can't see any other possibility to align actions within `AlertDialog`.

**Comment by HansMuller** [comment]
https://github.com/flutter/flutter/pull/85050 adds a MainAxisAlignment property to OverflowBar.

@kenzieschmoll, @ycherniavskyi - If we add an AlertDialog pass-through property for its OverflowBar's alignment, I think we can resolve this issue.

**Comment by ycherniavskyi** [comment]
@HansMuller it sounds great, but it seems that the redundant wrapper Container widget also could be removed in addition to the pass-through property for its OverflowBar's alignment.

Actually, here is my dirty temp workaround for alignment of AlertDialog actions:
```dart
diff --git a/packages/flutter/lib/src/material/dialog.dart b/packages/flutter/lib/src/material/dialog.dart
index 99d22ca161..063a7825bc 100644
--- a/packages/flutter/lib/src/material/dialog.dart
+++ b/packages/flutter/lib/src/material/dialog.dart
@@ -540,7 +540,7 @@ class AlertDialog extends StatelessWidget {
       actionsWidget = Padding(
         padding: actionsPadding,
         child: Container(
-          alignment: AlignmentDirectional.centerEnd,
+          alignment: AlignmentDirectional.center,
           padding: EdgeInsets.all(spacing),
           child: OverflowBar(
             spacing: spacing,
```

**Comment by ycherniavskyi** [comment]
@HansMuller also be great if AlertDialog actions alignment could be configured via some `Theme` property.

**Comment by HansMuller** [comment]
@ycherniavskyi - That's a [good idea](https://github.com/flutter/flutter/issues/69708#issuecomment-867413308), probably best to create a DialogTheme or an AlertDialogTheme. Will consider that once #85260 lands.
